tests/packages/mlx/test_mellanox-modules-compat.py

```diff
@@ -0,0 +1,19 @@
+# Requirements:
+# From --hosts parameter:
+# - host(A1): any master host of a pool, with access to XCP-ng RPM repositories and reports.xcp-ng.org.
+import logging
+from pkgfixtures import host_with_saved_yum_state
+
+def test_install_mlx_modules_alt(host_with_saved_yum_state):
```

Comment:
`host_with_saved_yum_state` will only restore the saved state after all tests of the package that need it have run. This means that it will not restore the state between your two tests. We don't have a fixture which does this at the test level, currently, but maybe we should. The problem is, our current implementation doesn't allow stacking saved yum states (and this sometimes allows us to detect weird overlaps of fixtures).

Reply:
Now that I think of it, nothing prevents us from restoring the yum state at the end of each test, but we must do so without discarding the saved state established by the `host_with_saved_yum_state` fixture before the first test.

Reply:
Following this idea, in `lib/hosts.py`, 

```
def yum_restore_saved_state(self):
```

could become

```
def yum_restore_saved_state(self, discard_saved_state=True):
```

Reply:
or I can also just uninstall the `mlx4-modules-alt` at the end of the first test...

Reply:
I mean yum insall and yum remove within the test itself. So I save the state and within the test I install/remove what I need and at the end the state is restored...